crates/swc_ecma_parser/src/parser/expr.rs

```diff
@@ -39,34 +39,32 @@ impl<I: Tokens> Parser<I> {
     pub(super) fn parse_assignment_expr(&mut self) -> PResult<Box<Expr>> {
         trace_cur!(self, parse_assignment_expr);
 
-        if self.input.syntax().typescript() && self.input.syntax().jsx() {
+        if self.input.syntax().typescript() && is!(self, JSXTagStart) {
```

Comment:
Does this mean we now produce `Token::JSXTagStart` even when `jsx()` is not enabled?

Reply:
Actually, when we encounter `JSXTagStart`, it indicates that JSX is enabled. The current implementation has just one fewer condition compared to the previous code, but logically, it should be equivalent in functionality.

This code ended up in its current state due to my rebase and squash process. However, upon reviewing it, I realize this part doesn’t actually need to be changed. If needed, I’m happy to revert this change. Please let me know your thoughts!

Reply:
I think the previous check is needless, as the existence of a such token proves `self.input.syntax().jsx() == true`. Thank you for clarification!